skimage/filters/edges.py

```diff
@@ -684,11 +684,30 @@ def laplace(image, ksize=3, mask=None):
     skimage.restoration.uft.laplacian().
 
     """
+    def k_size(ksize):
+        if ksize>31:
+            return "Kernel_size Must be less then 31"
+        if ksize%2==0:
+            return "Please use Odd  Kernel_Size"
+        if ksize==3:
+            param=[1,-2,1]
+            return param
+        else:
+            return np.convolve(k_size(ksize-2),[1,-2,1])
```

Comment:
Repeated convolution with [-1, 2, -1] was a mistake in my initial description of the issue. Each time we repeat the convolution we are taking another two derivatives! What we want is a ksize (central) finite different approximation to the 2nd derivative.

There are some specific cases (and a general algorithm) listed here:
https://en.wikipedia.org/wiki/Finite_difference_coefficient#Central_finite_difference
but OpenCV seems to be using a different approximation. We don't necessarily need to match OpenCV, though.

Reply:
okay thanks i will do more research and change it